contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/HomeMultiAMBErc20ToErc677.sol

```diff
@@ -273,20 +278,25 @@ contract HomeMultiAMBErc20ToErc677 is BasicMultiAMBErc20ToErc677, HomeFeeManager
     * @param _from address of sender, if bridge operation fails, tokens will be returned to this address
     * @param _receiver address of receiver on the other side, will eventually receive bridged tokens
     * @param _value bridged amount of tokens
+    * @return id of the created and passed message
     */
-    function passMessage(ERC677 _token, address _from, address _receiver, uint256 _value) internal {
+    function passMessage(ERC677 _token, address _from, address _receiver, uint256 _value) internal returns (bytes32) {
         bytes4 methodSelector = this.handleBridgedTokens.selector;
         address foreignToken = foreignTokenAddress(_token);
         bytes memory data = abi.encodeWithSelector(methodSelector, foreignToken, _receiver, _value);
 
-        bytes32 _messageId = bridgeContract().requireToPassMessage(
-            mediatorContractOnOtherSide(),
-            data,
-            requestGasLimit()
-        );
+        address executor = mediatorContractOnOtherSide();
+        uint256 gasLimit = requestGasLimit();
+        IAMB bridge = bridgeContract();
+
+        bytes32 _messageId = oracleDrivenLaneAllowed(foreignToken, _from, _receiver)
```

Comment:
Yes, setting a rule for unbridged tokens looks like a benefit for me. I think adding an explicit foolproof `require` statement makes sense, if the default destination lane (if destination lanes are not set for sender/receiver/token, use the manual lane). However, now, the default behaviour is to use an oracle-driven lane, so there won't be a way to block the token until it first bridged. Don't you think that this can be a downside of a such check?

Reply:
The functionality to forward messages operating with the particular pair of tokens to the manual lane is needed to stop attacks to dry out the validators' account balance. So, the scenario could be the following: someone create a fake erc20 token in the mainnet and transfer a billion of tokens to the xDai chain, then they relay small amounts of the token back to the Ethereum Mainnet. We don't know in advance which token will be transferred from the Mainnet that's why I think that the scenario to block a token which is not bridged yet is rare. But we definitely would have a motivation to handle a particular bridged token as so the oracles will not send signatures to the Ethereum Mainnet to finish the relay operation.

Anyway, it is not a big deal if you think that the lane for the tokens relaying operation must be identified by the token address on the foreign side. Just document this properly as so the code reviewers (e.g. security auditors) in the future could understand our reasoning. So, add the comment here and in the method `setTokenDestinationLane`.